src/Mod/Sketcher/Gui/TaskSketcherElements.cpp

```diff
@@ -273,76 +518,152 @@ TaskSketcherElements::TaskSketcherElements(ViewProviderSketch *sketchView)
     const char* ctrlKey = "Ctrl";
     QString cmdKey = QShortcut::tr(ctrlKey);
 #endif
-    QString zKey = QString::fromLatin1("Z");
-    ui->Explanation->setText(tr("<html><head/><body><p>&quot;%1&quot;: multiple selection</p>"
-                                "<p>&quot;%2&quot;: switch to next valid type</p></body></html>")
-                             .arg(cmdKey, zKey));
+
     ui->listWidgetElements->setSelectionMode(QAbstractItemView::ExtendedSelection);
     ui->listWidgetElements->setEditTriggers(QListWidget::NoEditTriggers);
     ui->listWidgetElements->setMouseTracking(true);
 
+    createSettingsButtonActions();
+
     // connecting the needed signals
     QObject::connect(
-        ui->listWidgetElements, SIGNAL(itemSelectionChanged()),
-        this                     , SLOT  (on_listWidgetElements_itemSelectionChanged())
+        ui->listWidgetElements, SIGNAL(itemPressed(QListWidgetItem *)),
+        this                     , SLOT  (on_listWidgetElements_itemPressed(QListWidgetItem *))

```

Comment:
I matched pre-existing formating. Should we replace those @abdullahtahiriyo ?

Reply:
I can see your point.

I agree that we should not use connectSlotsByName. We do not here. 

We still need to provide a name for the function. It happens that the name used by the "connectSlotsByName" function is incredibly convenient. It conveys the intention beautifully and we got already used to using it.

My impression is that we have that function naming quite extensively. I am wary of changing it, unless we decide something specific project-wide (such as removing the underscores while leaving the wording structure).

I think we should discuss this in the forum to see what others think. Maybe some are already using a given strategy.  I do not think we should change this right now.

Reply:
Hmmm... It is very dangerous situation IMO. By luck here we use a proxy widget to setup the UI. But let's imagine it moves in the future and the class becomes the direct owner of the UI (as is the case for example for `SketcherGeneralWidget`).
Then one will call `ui->setupUi(this)` and here is the mess. Because of the code generated by UIC, `setupUi()` will automatically call `QMetaObject::connectSlotsByName(this)` and so you'll get duplicate connection, that may be lead to very weird effects and be a PITA to debug. ;)

Reply:
You are totally right.

Although we should not reject different names for signals, I would be very much comfortable if we had a guideline saying, do not use the syntax connectSlotsByName uses, yet favour a specific naming convention (like having the connectSlotsByName way, but without underscores, or something similar). That makes the code much more readable IMO.

But that is aside of this PR.

Reply:
> yet favour a specific naming convention (like having the connectSlotsByName way, but without underscores, or something similar). That makes the code much more readable IMO.

I think this is what is already done (generally) in FC. So it's safe to go this way I think.
> But that is aside of this PR.

Got it. :)